Make use of TEXINPUTS from environment:  Fixes #106

mjsossong opened the pull request:
## Summary

I added g:Tex_TEXINPUTS to the TEXINPUTS in the run_cmd for the compiler and the bibtex compiler.

This initially failed with: `E121: Undefined variable: g:Tex_TEXINPUTS`, but that was resolved by adding
`let g:Tex_TEXINPUTS=$TEXINPUTS` to my .vimrc

To resolve this, I added defaulting of g:Tex_TEXINPUTS to '' if it was not defined.

## Other Information
This appears to be working properly now, both with and without g:Tex_TEXINPUTS being set.  If the user wants to make use of this functionality, they will have to add `let g:Tex_TEXINPUTS=$TEXINPUTS` to their .vimrc.

badouralix reviewed (commented):
That is a really nice idea, thanks for submitting it 🙇 

Maybe we should use `livepreview_texinputs` instead of `Tex_TEXINPUTS`, that would be more consistent with the existing configuration variables available in https://github.com/xuhdev/vim-latex-live-preview/blob/32d1a91/README.md#configuration
Slightly related, let's update the readme to document this new feature 

We have a standard way of initializing variables in the `Initialize()` function, let's leverage it https://github.com/xuhdev/vim-latex-live-preview/blob/32d1a911641e08c11b3f82ba886540cbae0b7513/plugin/latexlivepreview.vim#L282-L289

How about reordering TEXINPUTS to have this override at the very beginning ? I guess this is what we would expect when setting an override : 
1. Lookup in the static folders ( _e.g._ `s:static_texinput` )
2. Lookup in the buffer folder ( _i.e._ `l:tmp_root_dir` ) 
3. Lookup in the project folder ( _i.e._ `b:livepreview_buf_data['root_dir']` )
4. Lookup in the standard folders ( _i.e._ the final `:` )

```vim
'TEXINPUTS=' . s:static_texinput
             . ':' . l:tmp_root_dir
             . ':' . b:livepreview_buf_data['root_dir']
             . ': ' .
```

mjsossong commented:
Thanks for the feedback.  I've made all the suggested changes, and changed it to default to $TEXINPUTS.  I also added documentation to README.md.  Let me know what you think.

badouralix commented on README.md:
````diff
@@ -115,6 +115,17 @@ overridden by setting `g:livepreview_engine` variable:
+by setting the `g:livepreview_texinputs` variable:
+
+```vim
+let g:livepreview_engine = '/path1/to/files//:/path2/to/files//'
+```
+Note:  The double trailing `/` tells the compiler to search subdirectories.
````
````suggestion
let g:livepreview_texinputs = '/path1/to/files//:/path2/to/files//'
```

Note:  The double trailing `/` tells the compiler to search subdirectories.
````

badouralix commented on plugin/latexlivepreview.vim:
```diff
@@ -282,6 +283,9 @@ EEOOFF
     " Get the tex engine
     let s:engine = s:ValidateExecutables('livepreview_engine', [get(g:, 'livepreview_engine', ''), 'pdflatex', 'xelatex'])
 
+    " Initialize texinputs directory list to environment variable TEXINPUTS if g:livepreview_texinputs is not set
+    let s:static_texinputs = get(g:, 'livepreview_texinputs',$TEXINPUTS)
+
     " Get the previewer
     let s:previewer = s:ValidateExecutables('livepreview_previewer', [get(g:, 'livepreview_previewer', ''), 'evince', 'okular'])
```
Let's reorder these to keep the alphabetical order

```suggestion
    " Get the previewer
    let s:previewer = s:ValidateExecutables('livepreview_previewer', [get(g:, 'livepreview_previewer', ''), 'evince', 'okular'])

     " Initialize texinputs directory list to environment variable TEXINPUTS if g:livepreview_texinputs is not set
    let s:static_texinputs = get(g:, 'livepreview_texinputs', $TEXINPUTS)
```

badouralix commented on README.md:
````diff
@@ -115,6 +115,17 @@ overridden by setting `g:livepreview_engine` variable:
 ```vim
 let g:livepreview_engine = 'your_engine' . ' [options]'
 ```
+### TeX Inputs
````
```suggestion

### TeX Inputs
```

badouralix commented on plugin/latexlivepreview.vim:
```diff
@@ -75,7 +75,6 @@ function! s:Compile()
                 \ has_key(b:livepreview_buf_data, 'preview_running') == 0
         return
     endif
-
```
Should we remove this line ?

badouralix reviewed (commented):
Overall lgtm 🎉 
Added a few nitpicks

badouralix reviewed (approved):
LGTM :shipit:
